src/games/strategy/engine/framework/startup/ui/MetaSetupPanel.java

```diff
@@ -248,16 +248,12 @@ private static LobbyServerProperties getLobbyServerProperties() {
   private static Map<String, Object> matchCurrentVersion(List<Map<String, Object>> lobbyProps) {
     checkNotNull(lobbyProps);
     Version currentVersion = ClientContext.engineVersion().getVersion();
-    for (Map<String, Object> props : lobbyProps) {
-      if (props.containsKey("version")) {
-        Version otherVersion = new Version((String) props.get("version"));
-        if (otherVersion.equals(currentVersion)) {
-          return props;
-        }
-      }
-    }
 
-    return lobbyProps.get(0);
+    Optional<Map<String, Object>> matchingVersionProps = lobbyProps.stream()
+        .filter(props -> props.containsKey("version"))
+        .filter(props -> currentVersion.equals(props.get("version")))
```

Comment:
Those two lines could br merged into a single one... If get("version") returns null, null is always not equals to currentVersion